bokehjs/src/lib/models/widgets/tables/cell_formatters.ts

```diff
@@ -81,6 +81,63 @@ export class StringFormatter extends CellFormatter {
   }
 }
 
+export namespace ScientificFormatter {
+  export type Attrs = p.AttrsOf<Props>
+
+  export type Props = StringFormatter.Props & {
+    precision: p.Property<number | 10>
+    power_limit_high: p.Property<number>
+    power_limit_low: p.Property<number>
+  }
+}
+
+export interface ScientificFormatter extends ScientificFormatter.Attrs {}
+
+export abstract class ScientificFormatter extends StringFormatter {
+  properties: ScientificFormatter.Props
+
+  constructor(attrs?: Partial<ScientificFormatter.Attrs>) {
+    super(attrs)
+  }
+
+  static init_ScientificFormatter(): void {
+    this.define<ScientificFormatter.Props>({
+      precision:        [ p.Any,     10     ],
+      power_limit_high: [ p.Number,  5      ],
+      power_limit_low:  [ p.Number,  -3     ],
+    })
+  }
+
+  get scientific_limit_low(): number {
+    return Math.pow(10.0, this.power_limit_low)
+  }
+
+  get scientific_limit_high(): number {
+    return Math.pow(10.0, this.power_limit_high)
+  }
+
+  doFormat(row: any, cell: any, value: any, columnDef: any, dataContext: any): string {
+    let need_sci = value >= this.scientific_limit_high || value <= this.scientific_limit_low
```

Comment:
`needs_sci` must be `const`. Also, please reorder this condition, so that the arithmetic order is preserved, i.e.:
```js
value <= this.scientific_limit_low || this.scientific_limit_high <= value
```

Reply:
ok, I just noticed the test failure for the let, will change. The arithmetic order is copied from BasicTickFormatter, but I will rearrange it here